plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/EncoderServer.kt

```diff
@@ -118,10 +118,12 @@ class EncoderServer(val project: Project) : Disposable {
         val map = mutableMapOf<String, String>(
             "PORT" to port.toString(),
             "START_AMAZONQ_LSP" to "true",
-            "Q_WORKER_THREADS" to threadCount.toString(),
             "CACHE_DIR" to cachePath.toString(),
             "MODEL_DIR" to cachePath.resolve("qserver").toString()
         )
+        if (threadCount > 0 && threadCount < 100) {
```

Comment:
it looks like we already have input validation in the settings pane and i would rather have the constraints be defined in one place
```suggestion
        if (threadCount > 0) {
```